fullstack-templates/saas/turbowatch.ts

```diff
@@ -1,4 +1,73 @@
 import { defineConfig } from 'turbowatch';
+import { ChildProcess, exec, spawn } from 'child_process';
+
+// Function to execute a shell command and return a promise
+function executeCommand(command: string | string[], abortSignal?: AbortSignal): Promise<void> {
+    return new Promise((resolve, reject) => {
+        let process: ChildProcess;
+
+        if (typeof command === 'string') {
+            // Execute a simple command
+            process = exec(command, {cwd: __dirname}, (error, stdout, stderr) => {
+                if (error) {
+                    console.error(`exec error: ${error}`);
+                    return reject(error);
+                }
+                console.log(stdout);
+                console.error(stderr);
+                resolve();
+            });
+        } else {
+            // Execute a command with parameters (like 'cargo shuttle run')
+            process = spawn(command[0], command.slice(1), {
+                cwd: __dirname,
+                stdio: 'inherit',
+            });
+
+            process.on('close', (code) => {
+                if (code !== 0) {
+                    reject(new Error(`${command.join(' ')} exited with code ${code}`));
+                } else {
+                    resolve();
+                }
+            });
+        }
+
+        if (abortSignal) {
+            abortSignal.addEventListener('abort', () => {
+                process.kill(); // Kill the process if the watcher is aborted
+            });
+        }
+    });
+}
+
+let shuttleRuntimeAvailable = false;
```

Comment:
small nitpick: I would call it `shuttleCli` or `cargo-shuttle`, since the shuttle runtime is a different thing (a library that wraps the user app's main function).

Reply:
Thanks for the information I will rename it right away